pkg/promtail/targets/gcplog/formatter.go

```diff
@@ -0,0 +1,85 @@
+package gcplog
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/promtail/api"
+	"github.com/prometheus/common/model"
+)
+
+// LogEntry that will be written to the pubsub topic.
+// According to the following spec.
+// https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry
+type GCPLogEntry struct {
+	LogName  string `json:"logName"`
+	Resource struct {
+		Type   string            `json:"type"`
+		Labels map[string]string `json:"labels"`
+	} `json:"resource"`
+	Timestamp string `json:"timestamp"`
+
+	// The time the log entry was received by Logging.
+	// Its important that `Timestamp` is optional in GCE log entry.
+	ReceiveTimestamp string `json:"receiveTimestamp"`
+
+	TextPayload string `json:"textPaylod"`
+
+	// NOTE(kavi): There are other fields on GCPLogEntry. but we need only need above fields for now
+	// anyway we will be sending the entire entry to Loki.
+}
+
+func format(m *pubsub.Message, other model.LabelSet, useIncomingTimestamp bool) (api.Entry, error) {
+	var ge GCPLogEntry
+
+	if err := json.Unmarshal(m.Data, &ge); err != nil {
```

Comment:
Can you use jsoniter here instead of default json ? There's an ocean in performance difference :P https://github.com/json-iterator/go

Reply:
Interesting. Yea saw your bench marks on other PR :) Sure we can change that!

Reply:
Done 👍